src/DependencyInjection/FOSHttpCacheExtension.php

```diff
@@ -48,7 +48,7 @@ public function load(array $configs, ContainerBuilder $container): void
 
         if ($config['debug']['enabled'] || (!empty($config['cache_control']))) {
             $debugHeader = $config['debug']['enabled'] ? $config['debug']['header'] : false;
-            $ttlHeader = $config['cache_control']['ttl_header'];
+            $ttlHeader = $config['cache_control']['ttl_header'] ?? null;
```

Comment:
arf I was unsure so I wanted to keep `''` but at the end we removed it 
https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/pull/638#discussion_r1907479582

if null is ok here, and the confirmation will swap with default value, its perfect then 👍🏻

Reply:
This is strange it should be always `X-Reverse-Proxy-TTL` as in https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/blob/7b737f770941bfb67e065d1be5975aa26a3ee2d7/src/DependencyInjection/Configuration.php#L280 defined.

Looks for me there is an issue in applying the correct defaults from the config. /cc @dbu

Reply:
Looks for me in:

```
    ->children()
                ->arrayNode('cache_control')
```

The `->addDefaultsIfNotSet()` is missing.

But this makes the `(!empty($config['cache_control']))` check unnecessary or need to be changed to something else.

In case of Sulu `fos_http_cache.event_listener.cache_control` was never registered under `prod` environment. So to keep things as they were I think the check need to be changed. 

To keep things as they are we could remove `->defaultValue('X-Reverse-Proxy-TTL') ` and go here with `$ttlHeader = $config['cache_control']['ttl_header'] ?? 'X-Reverse-Proxy-TTL';`

Reply:
`$config['cache_control']['ttl_header'] ?? 'X-Reverse-Proxy-TTL';` was even my first idea, after we went for nothing on the original PR

yes perhaps in the meantime we can just fallback in the extension with same value as configuration

Reply:
IMHO `addDefaultsIfNotSet()` should be the correct way 🤔

Reply:
but its not an array node as per the doc

Reply:
to me the better "hot fix" would be to do like [initially done](https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/pull/638#discussion_r1907315992)

wdyt?

Reply:
but that would register always the service which is currently in none configured `cache_control` not there. For BC reasons we should not go with `addDefaultsIfNotSet`.

Reply:
Example we could go if we want the `addDefaultsIfNotSet` the `!empty` would need to be changed to:

```php
        if ($config['debug']['enabled']
           || count($config['cache_control']) > 1 // more as the ttl_header name was defined as that has a defaultValue and is always set
        ) {
            $debugHeader = $config['debug']['enabled'] ? $config['debug']['header'] : false;
            $ttlHeader = $config['cache_control']['ttl_header'];
```